playbooks/adhoc/s3_registry/s3_registry.yml

```diff
@@ -5,8 +5,10 @@
 #
 # The AWS access/secret keys should be the keys of a separate user (not your main user), containing only the necessary S3 access role.
 # The 'clusterid' is the short name of your cluster.
+#
+# This playbook should be limited to one openshift master
 
-- hosts: tag_clusterid_{{ clusterid }}:&tag_host-type_openshift-master
+- hosts: tag_clusterid_{{ clusterid }}:&tag_host-type_openshift-master[0]
```

Comment:
I admit I haven't played around too much with the more advanced ansible pattern matching here, but it looks to me like this might take the first host out of all with tag_host-type_openshift-master and then 'and' it with tag_cluster_id{{ clusterid }}, which would present issues with trying to run this with multiple clusters defined.